utils/test_harness/sysman/src/test_harness_sysman_power.cpp

```diff
@@ -44,29 +44,43 @@ zes_power_properties_t get_power_properties(zes_pwr_handle_t pPowerHandle) {
             zesPowerGetProperties(pPowerHandle, &pProperties));
   return pProperties;
 }
-uint32_t get_power_limit_count(zes_pwr_handle_t hPower) {
-  uint32_t pCount = 0;
-  EXPECT_EQ(ZE_RESULT_SUCCESS, zesPowerGetLimitsExt(hPower, &pCount, nullptr));
-  EXPECT_GE(pCount, 0);
-  return pCount;
+ze_result_t get_power_limit_count(zes_pwr_handle_t hPower, uint32_t *pCount) {
+  *pCount = 0;
+  ze_result_t status = zesPowerGetLimitsExt(hPower, pCount, nullptr);
+  if (status != ZE_RESULT_ERROR_UNSUPPORTED_FEATURE) {
```

Comment:
Can we return the status directly and move these checks to the caller ?
Hiding this detail under the harness looks easy to miss for the test writer

Reply:
`get_power_limit_count` is called by another harness method `get_power_limits_ext`. It is not called directly by the TESTS

Reply:
> `get_power_limit_count` is called by another harness method `get_power_limits_ext`. It is not called directly by the TESTS

Hmm .. That is still risky.
Please create another harness

Reply:
removed these checks from harness and added in actual TEST